### PR TITLE
Split mina_graphql.ml to a set of files

### LINKED ISSUE #13918: Exception thrown in zkapp tx generator
When launching tx generation with `max-cost: false`, I see the following two exceptions output:

```
2023-08-22 18:12:59.129716+02:00 Error ("Exception raised to [Monitor.try_with] that already returned.""
This error was captured by a default handler in [Async.Log]."(exn(
monitor.ml.Error
"Base_quickcheck.Generator.of_list: empty list"
("Raised at Base__Error.raise in file \"src/error.ml\" (inlined), line 8, characters 14-30"
"Called from Base__Error.raise_s in file \"src/error.ml\", line 9, characters 19-40"
"Called from Base_quickcheck__Generator.of_list in file \"src/generator.ml\", line 108, characters 7-79"

"Called from Mina_generators__Zkapp_command_generators.gen_account_update_body_components.(fun) in file \"src/lib/mina_generators/zkapp_command_generators.ml\", line 780, characters 12-46"

"Called from Base_quickcheck__Generator.bind.(fun) in file \"src/generator.ml\", line 66, characters 13-18"
"Called from Base_quickcheck__Generator.bind.(fun) in file \"src/generator.ml\", line 65, characters 12-36"
"Called from Base_quickcheck__Generator.bind.(fun) in file \"src/generator.ml\", line 65, characters 12-36"
"Called from Base_quickcheck__Generator.bind.(fun) in file \"src/generator.ml\", line 65, characters 12-36"

"Called from Mina_graphql.Mutations.Itn.schedule_zkapp_commands.(fun).go.(fun) in file "src/lib/mina_graphql/mina_graphql.ml\", line 4931, characters 38-1023"

"Called from Base__Result.try_with in file \"src/result.ml\", line 195, characters 9-15""

Caught by monitor Cohttp_async.Server.collect_errors"))))
```
and:
```
2023-08-22 18:12:59.129737+02:00 Error ("Exception raised to [Monitor.try_with] that already returned."
"This error was captured by a default handler in [Async.Log]."(exn(monitor.ml.Error(Failure
"timing task `itn_send_zkapp_commands` failed, exception reported to parent monitor")
("Raised at Stdlib.failwith in file \"stdlib.ml\", line 29, characters 17-33"
"Called from O1trace.exec_thread in file \"src/lib/o1trace/o1trace.ml\", line 77, characters 6-27"

"Called from Mina_graphql.Mutations.Itn.schedule_zkapp_commands.(fun).go in file \"src/lib/mina_graphql/mina_graphql.ml\", line 4893, characters 32-1023"

"Called from Async_kernel__Deferred0.bind.(fun) in file \"src/deferred0.ml\", line 54, characters 64-69"
"Called from Async_kernel__Job_queue.run_job in file \"src/job_queue.ml\" (inlined), line 128, characters 2-5"
"Called from Async_kernel__Job_queue.run_jobs in file \"src/job_queue.ml\", line 169, characters 6-47"

"Caught by monitor Cohttp_async.Server.collect_errors"))))                                                   
```

It happens on an attempt to send the third transaction.

Orchestrator script (executed with 10 private keys in `maxcost-keys`):

```json
{"action":"load-keys","params":{"dir":"maxcost-keys"}}
{"action":"zkapp-txs","params":{"experimentName":"exp13-9","tps":0.02,"minTps":0.01,"durationMin":10,"maxFee":4000000000,"minFee":2000000000,"zkapps":2,"newAccounts":0,"queueSize":5,"gap":0,"noPrecondition":false,"maxCost":false,"minBalanceChange":10,"maxBalanceChange":100000,"deploymentFee":2000000000,"feePayers":{"type":"output","step":-1,"name":"key"},"nodes":["127.0.0.1:11111"]}}
```

### PR DESCRIPTION
**Pull request by georgeee**
Twin against `develop`: [PR](https://github.com/MinaProtocol/mina/pull/14037).

File `mina_graphql.ml` is extremely large and poorly maintainable. Some code in it had around 10 layers of identation.

In this PR I make the first step to make this file more maintainable. A few modules are extracted out of `Mina_graphql` module.

Some structural changes were made to code in `Itn_zkapps.send_zkapps` function (compared to version before the code was moved). The only change to behavior is that an exception that would go unhandled now is being caight in `Itn_zkapps.send_zkapps`.

Except for `Itn_zkapps` module, all of the code was just moved. Some abstract types in `Types` module (like `'context`) were removed with concrete types to satisfy type checking.

Explain how you tested your changes:
* Tested this code on a cluster. No glitches were found.

Checklist:

- [x] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [x] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [x] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [x] All tests pass (CI will check this if you didn't)
- [x] Serialized types are in stable-versioned modules
- [x] Does this close issues? 

* Closes #13918

### CODE REVIEW

**Comment by psteckler** [comment]
!ci-build-me

File: src/lib/mina_graphql/types.ml [hunk 1]
```diff
@@ -0,0 +1,3229 @@
...
+        ~split:(fun f (t : input) -> f t.payment t.added_time)
+        ~fields:
+          [ arg "payment"
+              ~doc:(Doc.bin_prot "Serialized payment")
+              ~typ:(non_null string)
+          ; (* TODO: create a formal method for verifying that the provided added_time is correct  *)
+            arg "added_time" ~typ:(non_null string)
+              ~doc:
+                (Doc.date
+                   "Time that a payment gets added to another clients \
```
Comment (psteckler) [inline src/lib/mina_graphql/types.ml:2842]: client's

File: src/lib/mina_graphql/types.ml [hunk 2]
```diff
@@ -0,0 +1,3229 @@
...
+          [ arg "publicKey" ~typ:PublicKey.arg_typ
+              ~doc:
+                (sprintf
+                   "Public key you wish to start snark-working on; null to \
+                    stop doing any snark work. %s"
+                   Cli_lib.Default.receiver_key_warning )
+          ]
+  end
+
+  module AddPaymentReceiptInput = struct
```
Comment (psteckler) [inline src/lib/mina_graphql/types.ml:2827]: this module is not used anywhere, can be deleted

File: src/lib/mina_graphql/types.ml [hunk 3]
```diff
@@ -0,0 +1,3229 @@
...
+                 (obj "TarFile" ~fields:(fun _ ->
+                      [ field "tarfile" ~typ:(non_null string) ~args:[]
+                          ~resolve:(fun _ basename -> basename)
+                      ] ) ) )
+            ~doc:"Tar archive containing logs"
+            ~args:Arg.[]
+            ~resolve:(fun (_ : Mina_lib.t resolve_info) -> Fn.id)
+        ] )
+
+  let add_payment_receipt =
```
Comment (psteckler) [inline src/lib/mina_graphql/types.ml:2002]: this definition is not used, can be deleted

File: src/lib/mina_graphql/doc.ml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+open Core_kernel
+
+let date ?(extra = "") s =
+  sprintf
+    "%s (stringified Unix time - number of milliseconds since January 1, \
+     1970)%s"
+    s extra
+
+let bin_prot = sprintf "%s (base58-encoded janestreet/bin_prot serialization)"
```
Comment (psteckler) [inline src/lib/mina_graphql/doc.ml:9]: can be deleted, only used in `AddPaymentReceiptInput`

**Review by psteckler — APPROVED** [review]
Approved, but see comments

**Comment by deepthiskumar** [comment]
!ci-build-me

File: src/lib/mina_graphql/misc.ml [hunk 5]
```diff
@@ -0,0 +1,44 @@
+open Core
```
Comment (deepthiskumar) [inline src/lib/mina_graphql/misc.ml:1]: nit: rename this to `utils.ml`?

**Comment by georgeee** [comment]
_Waiting for #14037 to be merged to unblock CI_

**Review by MartinMinkov — APPROVED** [review]

**Comment by georgeee** [comment]
!ci-build-me

**Comment by deepthiskumar** [comment]
!approved-for-mainnet

**Pull request merged**